models/classes/Copier/InstanceCopier.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2022 (original work) Open Assessment Technologies SA.
+ *
+ * @author Andrei Shapiro <andrei.shapiro@taotesting.com>
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoItems\model\Copier;
+
+use core_kernel_classes_Class;
+use core_kernel_classes_Resource;
+use oat\tao\model\resources\Contract\InstanceCopierInterface;
+use oat\tao\model\resources\Contract\InstanceContentCopierInterface;
+
+class InstanceCopier implements InstanceCopierInterface
+{
+    /** @var InstanceCopierInterface */
+    private $taoInstanceCopier;
+
+    /** @var InstanceContentCopier */
+    private $instanceContentCopier;
+
+    public function __construct(
+        InstanceCopierInterface $taoInstanceCopier,
+        InstanceContentCopierInterface $instanceContentCopier
+    ) {
+        $this->taoInstanceCopier = $taoInstanceCopier;
+        $this->instanceContentCopier = $instanceContentCopier;
+    }
+
+    public function copy(
+        core_kernel_classes_Resource $instance,
+        core_kernel_classes_Class $destinationClass
+    ): core_kernel_classes_Resource {
+        $newInstance = $this->taoInstanceCopier->copy($instance, $destinationClass);
+        $this->instanceContentCopier->copy($instance, $newInstance);
+
+        return $newInstance;
+    }
```

Comment:
See https://github.com/oat-sa/tao-core/pull/3503/files#r863544022: By changing the signature in `InstanceContentCopierInterface` to return the destination resource we may directly return the return value from the copy call. We may also use `$from` and `$to` fro parameter names (see https://github.com/oat-sa/tao-core/pull/3503/files#r863544022)

```suggestion
    public function copy(
        core_kernel_classes_Resource $from,
        core_kernel_classes_Class $to
    ): core_kernel_classes_Resource {
        return $this->instanceContentCopier->copy(
            $from,
            $this->taoInstanceCopier->copy($from, $to)
        );
    }
```

Reply:
Please, lets return void, cause the Resource object referenced is already the one that should be returned.